src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs

```diff
@@ -510,6 +520,54 @@ public async void UpdateReviewBackground()
                     _telemetryClient.StopOperation(operation);
                 }
             }
+        }
+
+        public async Task<CodeFile> GetCodeFile(string repoName, string buildId, string artifactName, string packageName, string originalFileName, string codeFileName, MemoryStream memoryStream)
+        {
+            Stream stream = null;
+            CodeFile codeFile = null;
+            if (string.IsNullOrEmpty(codeFileName))
+            {
+                // backward compatibility until all languages moved to sandboxing of codefile to pipeline
+                stream = await _devopsArtifactRepository.DownloadPackageArtifact(repoName, buildId, artifactName, originalFileName, "file");
+                codeFile = await CreateCodeFile(Path.GetFileName(originalFileName), stream, false, memoryStream);
+            }
+            else
+            {
+                stream = await _devopsArtifactRepository.DownloadPackageArtifact(repoName, buildId, artifactName, packageName, "zip");
```

Comment:
Will we require the file already be a zip or will devops already do that if we point it to an artifact?

Reply:
devops does it for us when we request download. We don't need to zip them in advance